2025/kirtanchandak/kirtanchandak-gsoc-2025-report.md

```diff
@@ -0,0 +1,224 @@
+  </video>
+</p>
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/openfoodfacts-webcomponents/pull/200
+
+### **User Role Management System**
+
+Developed a comprehensive user role management system for the folksonomy engine. This allows us to perform operations like bulk deletions or property modifications are restricted to trusted contributors.
+
+The role management system includes different user levels with appropriate permissions - `user`, `moderator` and `admin` helping maintain data quality while allowing community participation.
+
+Implemented API for checking user role `/user/me` on OpenFoodFacts -
+
+Documentation:
+
+- https://api.folksonomy.openfoodfacts.org/docs#/default/get_user_info_user_me_get
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/folksonomy_api/pull/295
+
+### **Property Exploration Tools**
+
+### WebComponents:
+
+Built comprehensive exploration tools for both folksonomy properties and folksonomy product properties. These tools allow users to browse, search, and analyze the existing properties in the system, making it easier to understand what data is available and how it's being used.
+
+The exploration tools include filtering capabilities, download CSV, and action like delete, rename for moderators and admins.
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/openfoodfacts-webcomponents/pull/181
+- https://github.com/openfoodfacts/openfoodfacts-webcomponents/pull/182
+
+### Integration
+
+Integrated these WebComponents into the Open Food Facts server, making the new tools available to all users.
+
+#### Folksonomy Product Properties -
+
+<p align="center">
+  <img src="./assets/folksonomy-product-properties.png" width="700" />
+</p>
+
+### Folksonomy Properties -
+
+<p align="center">
+  <img src="./assets/folksonomy-properties.png" width="700" />
+</p>
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/openfoodfacts-server/pull/12148
+- https://github.com/openfoodfacts/openfoodfacts-server/pull/12277
+
+### **Bulk Property/Value Management APIs**
+
+Implemented a robust bulk value management system designed to fight vandalism and maintain data quality. The system includes features for bulk removal, renaming, deletion, replacement, and clash detection of property values.
+
+This tool is essential for maintaining the integrity of the folksonomy-engine, allowing administrators to quickly address data quality issues, remove spam or incorrect entries. The clash detection feature helps prevent conflicts when merging or modifying values.
+
+Documantation:
+
+- https://api.folksonomy.openfoodfacts.org/docs#/Admin%20-%20Property%20Management
+- https://api.folksonomy.openfoodfacts.org/docs#/Admin%20-%20Value%20Management
+
+**Relevant PRs:**
+
+- https://github.com/openfoodfacts/folksonomy_api/pull/302
+
+### **Bulk Property/Value Management UI**
+
+Implemented UI for renaming, deleting and checking conflicts pefore deleting for values and properties.
```

Comment:
I'm reading quickly and find some typo, so I suggest doing a thorough review on your side to spot remaining typos.
```suggestion
Implemented UI for renaming, deleting and checking conflicts before deleting for values and properties.
```